File: .github/resources/convert_json_to_csv.js
const fs = require('fs');
const { parse } = require('json2csv');

const convertJsonToCsv = (jsonFilePath, csvFilePath) => {
  // Load JSON data
  const jsonData = JSON.parse(fs.readFileSync(jsonFilePath, 'utf-8'));

  // Debug: Print the loaded JSON data
  console.log("Loaded JSON data:", JSON.stringify(jsonData, null, 2));

  const fields = ['Target', 'VulnerabilityID', 'Severity', 'CVSS Score', 'Title', 'Library', 'Vulnerable Version', 'Fixed Version', 'Information URL'];
  const csvData = jsonData.Results.flatMap(result => {
    console.log(`Processing target: ${result.Target}`);  // Debug: Print the target being processed

    if (!result.Vulnerabilities) {
      console.warn(`No vulnerabilities found for target: ${result.Target}`);
      return [];
    }

    return result.Vulnerabilities.map(vulnerability => {
      console.log(`Processing vulnerability: ${vulnerability.VulnerabilityID}`);  // Debug: Print the vulnerability ID
      return {
        Target: result.Target,
        VulnerabilityID: vulnerability.VulnerabilityID || '',
        Severity: vulnerability.Severity || '',
        'CVSS Score': vulnerability.CVSS?.nvd?.V3Score || '',
        Title: vulnerability.Title || '',
        Library: vulnerability.PkgName || '',
        'Vulnerable Version': vulnerability.InstalledVersion || '',
        'Fixed Version': vulnerability.FixedVersion || '',
        'Information URL': vulnerability.PrimaryURL || ''
      };
    });
  });

  // Debug: Print the CSV data before writing
  console.log("CSV Data:", csvData);

  const csv = parse(csvData, { fields });
  fs.writeFileSync(csvFilePath, csv);
  console.log(`CSV file written to ${csvFilePath}`);
};

const [jsonFilePath, csvFilePath] = process.argv.slice(2);
convertJsonToCsv(jsonFilePath, csvFilePath);

File: .github/resources/convert_json_to_csv.py
import json
import csv
import argparse

def convert_json_to_csv(json_file_path, csv_file_path):
    # Load JSON data
    with open(json_file_path, 'r') as json_file:
        data = json.load(json_file)
    
    # Debug: Print the loaded JSON data
    print("Loaded JSON data:", json.dumps(data, indent=2))

    # Open CSV file for writing
    with open(csv_file_path, 'w', newline='') as csv_file:
        csv_writer = csv.writer(csv_file)

        # Write CSV header for vulnerabilities
        csv_writer.writerow(['Target', 'VulnerabilityID', 'Severity', 'CVSS Score', 'Title', 'Library', 'Vulnerable Version', 'Fixed Version', 'Information URL'])

        # Iterate over results and write vulnerabilities to CSV
        for result in data.get('Results', []):
            target = result.get('Target', '')
            print(f"Processing target: {target}")  # Debug: Print the target being processed
            for vulnerability in result.get('Vulnerabilities', []):
                print(f"Processing vulnerability: {vulnerability.get('VulnerabilityID', '')}")  # Debug: Print the vulnerability ID
                csv_writer.writerow([
                    target,
                    vulnerability.get('VulnerabilityID', ''),
                    vulnerability.get('Severity', ''),
                    vulnerability.get('CVSS', {}).get('nvd', {}).get('V3Score', ''),
                    vulnerability.get('Title', ''),
                    vulnerability.get('PkgName', ''),
                    vulnerability.get('InstalledVersion', ''),
                    vulnerability.get('FixedVersion', ''),
                    vulnerability.get('PrimaryURL', '')
                ])

        # Write CSV header for dependencies
        csv_writer.writerow(['Target', 'ID', 'Name', 'Version'])

        # Iterate over results and write dependencies to CSV
        for result in data.get('Results', []):
            target = result.get('Target', '')
            print(f"Processing target for dependencies: {target}")  # Debug: Print the target being processed for dependencies
            for package in result.get('Packages', []):
                print(f"Processing package: {package.get('ID', '')}")  # Debug: Print the package ID
                csv_writer.writerow([
                    target,
                    package.get('ID', ''),
                    package.get('Name', ''),
                    package.get('Version', '')
                ])

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Convert Trivy JSON scan results to CSV.')
    parser.add_argument('json_file', help='Path to the JSON file containing Trivy scan results.')
    parser.add_argument('csv_file', help='Path to the output CSV file.')

    args = parser.parse_args()

    convert_json_to_csv(args.json_file, args.csv_file)